vast/losses/losses.py

```diff
@@ -1,7 +1,6 @@
 import torch
 from torch.nn import functional as F
-
-from .. import tools
+from vast import tools
```

Comment:
I have undone this.

Reply:
Thank you @siebenkopf 
The change was made thinking it was much more readable and it was working for me.
Just so that I understand this correctly, `from vast import tools` would fail only if `vast` has not been installed with `pip` or not in your `PYTHONPATH`?
I can see another corner case problem here in case multiple versions of `vast` are present in which case this may import the version installed rather than the version in the current working directory.
I see many other repos like the official pytorch repo taking the approach `from torch.ao.quantization.fake_quantize import FakeQuantizeBase `.
Is my understanding of the above problem correct, because I think I am doing this at many other places in the codebase, which might need to be reverted?

Reply:
Generally: the package is defined in the `__init__.py` files in the directories. Only what you import there can be accessed trough your main package. For example, if you have the following structure:
```
package
- __init__.py
- code.py
```

with contents:
```
code.py:
def some_func():
  pass
```

```
__init__.py:
from . import code
```

Then you can call:
```
>>> import package
>>> package.code.some_func()
```

If you would leave the `__init__.py` empty, you would need to call:
``` 
>>> import package.code
>>> package.code.some_func()
```

Note that the `code.py` could as well be a directory called `code` with a file `code/__init__.py`, where the above rules apply recursively.

Now, when you are inside of a python package (a directory that contains an `__init__.py` file), you should never import your own package. This might end up in circular imports. For you, it was working so far since you did not include all your files into the package (in the `__init__.py` file you did not import objects from some files).

There is an exception: when you run stand-alone code, e.g. scripts that are not inside of any `__init__.py` file, you can use either relative imports (`from .. import`) or absolute (`from package import`). In that case, I usually prefer the absolute import since then you can just copy that scripts anywhere else and run it -- this is usually good when you write example code.

Reply:
Thankyou!